pkg/tsdb/grafana-testdata-datasource/scenarios.go

```diff
@@ -390,6 +396,31 @@ func (s *Service) handleCSVMetricValuesScenario(ctx context.Context, req *backen
 	return resp, nil
 }
 
+func (s *Service) handleQueryMetaScenario(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	resp := backend.NewQueryDataResponse()
+
+	if len(req.Queries) == 0 {
+		return nil, errors.New("no queries")
+	}
+
+	refId := req.Queries[0].RefID
+
+	username := req.PluginContext.User.Name
+
+	keys := []string{"username"}
+	values := []string{username}
```

Comment:
I know you only care about username right now, but maybe no harm in adding a few more for completeness.

For example (all of which can be found in PluginContext):
- Org ID
- User Role
- User Agent
- Plugin ID
- Plugin Version